### PR TITLE
Fixes IV Drips

### PR DESCRIPTION
**Pull request by Fox-McCloud**
So, apparently someone didn't understand reagents very well.

This checked for a reagent containers volume...which is always above 0. If it was 0, it wouldn't be a reagent container.

Secondly, when doing the fractionalizing, it would also go by the reagent containers volume, as opposed to the reagent's volume.

This means that a blood transfer would transfer 10 units on the first process, then it would do 9.5 on the second....then 9 on the third. You quickly run into diminishing returns...and the end result is that a 200 unit blood pack barely restores 100 units of blood.

Either case, this fixes that by ensuring the code doesn't run if there aren't any reagents in the reagent container attached to the, and it ensure the proper ratio for injecting is always maintained.

:cl: Fox McCloud
fix: Fixes IV drips not properly injecting the right amount of blood
/:cl:

### CODE REVIEW

File: code/game/machinery/iv_drip.dm [hunk 1]
```diff
@@ -117,13 +117,13 @@
...
-			if(beaker.volume > 0)
+			if(beaker.reagents.total_volume)
 				var/transfer_amount = 5
 				if(istype(beaker, /obj/item/weapon/reagent_containers/blood))
 					// speed up transfer on blood packs
 					transfer_amount = 10
-				var/fraction = min(transfer_amount/beaker.volume, 1) //the fraction that is transfered of the total volume
-				beaker.reagents.reaction(attached, INJECT, fraction,0) //make reagents reacts, but don't spam messages
+				var/fraction = min(transfer_amount/beaker.reagents.total_volume, 1) //the fraction that is transfered of the total volume
+				beaker.reagents.reaction(attached, INJECT, fraction, 0) //make reagents reacts, but don't spam messages
```
Comment (optimumtact) [inline code/game/machinery/iv_drip.dm]: TRUE/FALSE defines there?

**Review by optimumtact — APPROVED** [review]

File: code/game/machinery/iv_drip.dm (hunk 1 above)
Comment (Cyberboss) [inline code/game/machinery/iv_drip.dm]: 0 should be FALSE here

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by Fox-McCloud** [comment]
@optimumtact @Cyberboss 

Made the styling changes.

**Review by Cyberboss — APPROVED** [review]

File: code/game/machinery/iv_drip.dm [hunk 2]
```diff
@@ -192,7 +201,8 @@
 /obj/machinery/iv_drip/examine()
 	set src in view()
 	..()
-	if (!(usr in view(2)) && usr!=loc) return
+	if(!(usr in view(2)) && usr != loc)
```
Comment (optimumtact) [inline code/game/machinery/iv_drip.dm]: this is weird code

Comment (optimumtact) [inline code/game/machinery/iv_drip.dm]: set src in view() is also some oldcode nonsense afaik.

Comment (Fox-McCloud) [inline code/game/machinery/iv_drip.dm]: it's really weird, but I didn't really want to touch it, either.

Comment (Jalleo) [inline code/game/machinery/iv_drip.dm]: Its paprika code.

Comment (RemieRichards) [inline code/game/machinery/iv_drip.dm]: Change the start of this proc to:

```
/obj/machinery/iv_drip/examine(mob/user)
    ..()
    if(get_dist(user, src) > 2)
        return
```

```view()``` isn't necessary, as you can't examine something with out being able to see it, and the ```set src in x``` is garbage code and ```usr``` is too but we have ```user``` instead.

**Pull request merged**